app/src/main/java/ch/epfl/favo/MainActivity.java

```diff
@@ -223,22 +228,23 @@ public void onBackPressed() {
     DrawerLayout mDrawerLayout = findViewById(R.id.drawer_layout);
     if (mDrawerLayout.isDrawerOpen(GravityCompat.START))
       mDrawerLayout.closeDrawer(GravityCompat.START);
-    else {
+    else {/*
       NavHostFragment host =

```

Comment:
Why commenting this part? This would actually be useful in order to always come back to the home page when pressing the back button

Reply:
I tried and found it still works after commenting. Correct me if I misunderstand something

Reply:
No the intention was to always return to the home page whatever the history of pages you open (overwriting the default behavior in android). Try navigating to the account page, then directly to the settings page. If you press back from the settings page, you come back to the account page, not to the map page

Reply:
Ohh, sorry it's my careless decision. I must be desperate to comment some uncovered code to improve test coverage at the moment. It's my fault.

Reply:
reverted